experimental/devices/nrzled/nrz_spi.go

```diff
@@ -0,0 +1,453 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package nrzled
+
+import (
+	"errors"
+	"fmt"
+	"image"
+	"image/color"
+	"image/draw"
+
+	"periph.io/x/periph/conn/display"
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+)
+
+// SPIDev represents a strip of WS2812b LEDs as a strip connected over a SPI port.
+// It accepts a stream of raw RGB pixels and converts it to a bit pattern consistent
+// with the WS812b protocol.
+// Includes intensity and temperature correction.
+type SPIDev struct {
```

Comment:
Oh I was thinking about using Dev and switching based on if `p` or `s` is nil. The runtime cost would be trivial, and the switch would only be at two locations: String(), Draw() and Write(), but Draw and Write() could call a small utility function instead. That would remove a *ton* of code duplication.